### ISSUE TITLE
GLES 2 fallback not working on Desktop Safari

### ISSUE THREAD

**Issue by robgraeber** [godotengine/godot#26899]
**Godot version:** 3.1 rc2
**OS/device including version:** Latest OS X

**Issue description:** When compiling dodge the creep for GLES3 and enabling the GLES2 fallback in project settings, the game fails to load in desktop Safari with the following error: 
```
[Warning] exception thrown: TypeError: undefined is not an object (evaluating 'GLctx.getParameter'),_glGetString@file:///Users/bobby/Downloads/godot-demo-projects-master/2d/dodge_the_creeps/Dodge%20the%20Creeps.js:7:264749 (Dodge the Creeps.js, line 7)
wasm-stub@[wasm code]
<?>.wasm-function[__ZN15RasterizerGLES310initializeEv]@[wasm code]
<?>.wasm-function[__ZN18VisualServerRaster4initEv]@[wasm code]
<?>.wasm-function[__ZN13OS_JavaScript10initializeERKN2OS9VideoModeEii]@[wasm code]
<?>.wasm-function[__ZN4Main6setup2Ey]@[wasm code]
<?>.wasm-function[__ZN4Main5setupEPKciPPcb]@[wasm code]
<?>.wasm-function[_main]@[wasm code]
wasm-stub@[wasm code]
jj@[native code]
file:///Users/bobby/Downloads/godot-demo-projects-master/2d/dodge_the_creeps/Dodge%20the%20Creeps.js:7:316260
callMain@file:///Users/bobby/Downloads/godot-demo-projects-master/2d/dodge_the_creeps/Dodge%20the%20Creeps.js:7:319971
synchronousStart@file:///Users/bobby/Downloads/godot-demo-projects-master/2d/dodge_the_creeps/Dodge%20the%20Creeps.js:216:23
apply@[native code]
promiseReactionJob@[native code]
[Error] undefined is not an object (evaluating 'GLctx.getParameter')
	displayFailureNotice (Dodge the Creeps.html:241)
	promiseReactionJob
[Warning] **ERROR**: Condition ' !_start_success ' is true. returned: false (Dodge the Creeps.js, line 7)
[Warning]    At: main/main.cpp:1277:start() - Condition ' !_start_success ' is true. returned: false (Dodge the Creeps.js, line 7)
``` 

**Steps to reproduce:** Run dodge the creeps and enable the GLES2 fallback in project settings.

**Comment by akien-mga** [comment]
Does it work if you export GLES2 directly instead of using GLES3 + fallback?

**Comment by robgraeber** [comment]
Yeah GLES2 works if compiled directly, will post a lot tonight.

**Comment by akien-mga** [comment]
Did it use to work in previous releases? Especially prior to beta 11 where I reworked the fallback system with #26523 (and then #26633 in RC 1).

If not, it might just be that the fallback system never worked for JS. CC @hpvb @eska014

**Comment by robgraeber** [comment]
@akien-mga I tried compiling Dodge The Creeps with 3.1 beta 10 and it had the same issue

**Comment by akien-mga** [comment]
Thanks, so I didn't break it :D
It might never have worked in the first place, @eska014 will have to review/fix @hpvb's fallback code for HTML5.

**Comment by clayjohn** [comment]
Is this still reproducible in 3.2? 

It seems important to fix, but likely won't make it in time for 3.2. Pushing back to 3.2.1.

**Comment by KyleFromOhio** [comment]
had same issue in 3.2

**Comment by Zappo-II** [comment]
Can confirm, same for me...

> `undefined is not an object (evaluating 'GLctx.getParameter')`

```
$ godot --version
3.2.stable.official
```

Checked on:

* MACOSX 10.15.3 / Safari 13.05
* IOS 13.1 (17A844) / Safari

Interesting Details:

* On IOS if Enabling Experimental Features (WEBGPU / WEBGL2) of IOS Safari my Project will "somehow" load and do something (play jittered sound, react to touch input), but won't show anything on screen...
* Certainly my project IS working on common browsers on other OSs...
* It's also working on MACOS in non Safari Browsers (Chromium, ...)

**Comment by thomcraver** [comment]
Can confirm for my son's 3D projects, as well. HTML5 version works on the Linux and Windows - all browsers. Macbooks have issues in Chrome, Safari and FireFox. Intro screen plays. Start button can be clicked, next scene loads music, but nothing else. Dark gray screen. 

Same behavior with native MacOS. 

GL3, GL2 fallback.

**Comment by Calinou** [comment]
I don't think the GLES2 fallback was ever implemented for HTML5, or was it?

**Comment by triptych** [comment]
This is still happening
` undefined is not an object (evaluating 'GLctx.getParameter') ` 
Basic godot game https://musing-brattain-f39868.netlify.app/
Desktop Safari --> Version 13.1.1 (14609.2.9.1.3)
Model Name:	MacBook Pro
  Model Identifier:	MacBookPro14,3
  Processor Name:	Intel Core i7
  Processor Speed:	3.1 GHz
  Number of Processors:	1
  Total Number of Cores:	4
  L2 Cache (per Core):	256 KB
  L3 Cache:	8 MB
  Hyper-Threading Technology:	Enabled
  Memory:	16 GB
  Boot ROM Version:	207.0.0.0.0

**Comment by royalstream** [comment]
Just happened to me, using Godot 3.2.4 beta while trying to run the Platformer 2D demo HTML5 on my iPhone.
The error happens with iOS Safari and also with iOS Chrome.

**Comment by royalstream** [comment]
This was happening to me but I was mistakenly using GLES3 in the Project Settings.
As soon as I changed it to GLES2 it worked fine under Safari in both macOS and iOS. 
YMMV

**Comment by alexzheng** [comment]
Still happening in Godot 3.2.4 beta6.

**Comment by mrimvo** [comment]
Still happens in Godot 3.3 RC6, `fallback_to_gles2` enabled.
![](https://cdn.discordapp.com/attachments/788789549912424491/828690680661540945/unknown.png)

**Comment by akien-mga** [comment]
Fixed by #47659.
